src/apm_client.rs

```diff
@@ -69,11 +73,21 @@ impl ApmClient {
                 }
             })
             .map(Arc::new);
+        let client = if allow_invalid_certs {
+            let builder = reqwest::ClientBuilder::new().danger_accept_invalid_certs(true);
+
+            #[cfg(feature = "native-tls")]
```

Comment:
Looks like this call should be present regardless on chosen TLS backend.

Reply:
danger_accept_invalid_hostnames is only supported by native-tls https://docs.rs/reqwest/0.11.3/reqwest/struct.ClientBuilder.html#method.danger_accept_invalid_hostnames. Though this could probably be split into a separate bool if that is preferable.

Reply:
According to the docs, `danger_accept_invalid_hostnames` takes and returns `self `by value. This code should not successfully build. Also, there is no `native-tls` feature in the crate - there's only `default-tls`, which translates to the same reqwest feature.

Reply:
That flag is not directly in the scope of this PR I will remove it.